### PR TITLE
Add benchmark for distance functions

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Review by vekterli — DISMISSED** [review]
:+1:

File: searchlib/src/tests/tensor/distance_functions/distance_functions_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+size_t npos = std::string::npos;
+
+void run_calc(size_t iterations, TypedCells b, const BoundDistanceFunction & df) __attribute_noinline__;
+void run_calc_with_limit(size_t iterations, TypedCells b, const BoundDistanceFunction & df) __attribute_noinline__;
+
+void
+run_calc(size_t iterations, TypedCells b, const BoundDistanceFunction & df) {
+    vespalib::steady_time start = vespalib::steady_clock::now();
+    for (size_t i(0); i < iterations; i++) {
+        df.calc(b);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/tensor/distance_functions/distance_functions_benchmark.cpp]: printing and using in assert which should cover it.

File: searchlib/src/tests/tensor/distance_functions/distance_functions_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    TypedCells a_cells(av), b_cells(bv);
+
+    run_calc(iterations, b_cells, *df.for_query_vector(a_cells));
+    run_calc_with_limit(iterations, b_cells, *df.for_query_vector(a_cells));
+}
+
+template<typename T>
+void benchmark(size_t iterations, size_t elems, const std::string & dist_functions) {
+    if (dist_functions.find("euclid") != npos) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/tensor/distance_functions/distance_functions_benchmark.cpp:77]: At least compiler does not give me the option yet.

File: searchlib/src/tests/tensor/distance_functions/distance_functions_benchmark.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    if (argc > 2) { num_elems = atol(argv[2]); }
+    if (argc > 3) { dist_functions = argv[3]; }
+    if (argc > 4) { data_types = argv[4]; }
+
+    printf("Benchmarking %ld iterations with vector length %ld with distance functions '%s' for data types '%s'\n",
+           num_iterations, num_elems, dist_functions.c_str(), data_types.c_str());
+    benchmark(num_iterations, num_elems, dist_functions, data_types);
+
+    return 0;
+}
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/tensor/distance_functions/distance_functions_benchmark.cpp]: Fixed

**Pull request merged**